src/components/common/DropdownButton.test.js

```diff
@@ -0,0 +1,133 @@
+import { shallow } from 'enzyme';
+import React from 'react';
+import { DropdownToggle, DropdownItem } from 'reactstrap';
+import DropdownButton from './DropdownButton';
+
+const validDropdownItems = [
+  {
+    display: 'Nice',
+    value: 'banana',
+    icon: 'cubes',
+  },
+  {
+    display: 'Cool',
+    value: 'apple',
+    icon: 'mobile',
+  },
+  {
+    display: 'Fun',
+    value: 'pear',
+    icon: 'wind',
+  },
+];
+
+describe('DropdownButton', () => {
+  it('smoke test', () => {
+    const component = shallow(<DropdownButton DropdownItems={validDropdownItems} />);
+    expect(component.exists()).toBe(true);
+  });
+
+  it('renders every dropdown item correctly', () => {
+    const component = shallow(
+      <DropdownButton
+        icon="bacteria"
+        DropdownItems={validDropdownItems}
+        defaultOpen
+        displayValue="bar"
+      />,
+    );
+    expect(component).toMatchSnapshot();
+  });
+
+  it('handles defaultOpen properly', () => {
+    const componentTrue = shallow(
+      <DropdownButton
+        icon="lungs"
+        DropdownItems={validDropdownItems}
+        defaultOpen
+        displayValue="foo"
+      />,
+    );
+    expect(componentTrue).toMatchSnapshot();
+
+    const componentDefault = shallow(
+      <DropdownButton icon="lungs" DropdownItems={validDropdownItems} displayValue="foo" />,
+    );
+    expect(componentDefault).toMatchSnapshot();
+
+    const componentFalse = shallow(
+      <DropdownButton
+        icon="lungs"
+        DropdownItems={validDropdownItems}
+        defaultOpen={false}
+        displayValue="foo"
+      />,
+    );
+    expect(componentFalse).toMatchSnapshot();
+  });
+
+  it('handles dropdown click correctly', () => {
```

Comment:
OK, I don't think that the clicks are going through; I believe the clicks are handled in a way where shallow rendering would not allow that to happen. This could be fixed by mounting the component; however, I don't think this is necessary, as this is purely functionality implemented by the library anyway. I think this test can just be removed.